CryptoTechReminderSystem/UseCase/GetProjectManagersWithOpenTimeEntries.cs

```diff
@@ -0,0 +1,101 @@
+
+            return ProjectManagersWithOpenTimeEntries(timeSheets, billablePeople, slackGetBillablePeopleResponse);
+        }
+
+        private GetLateBillablePeopleResponse ProjectManagersWithOpenTimeEntries(IList<DomainObject.TimeSheet> timeSheets, IList<DomainObject.HarvestBillablePerson> billablePeople, IList<DomainObject.SlackBillablePerson> slackGetBillablePeopleResponse) {
+            var getProjectManagersWithOpenTimeEntriesResponse = new GetLateBillablePeopleResponse
+            {
+                BillablePeople = new List<GetLateBillablePeopleResponse.LateBillablePerson>()
+            };
+
+            foreach (var timeSheet in timeSheets) {
+                if (!timeSheet.IsClosed)
+                {
+                    foreach (var harvestProjectManager in HarvestProjectManagers(billablePeople, timeSheet))
+                    {
+                        var slackProjectManager =
+                            SlackProjectManager(slackGetBillablePeopleResponse, harvestProjectManager);
+                        
+                        if (slackProjectManager != null)
+                        {
+                            getProjectManagersWithOpenTimeEntriesResponse.BillablePeople.Add(new GetLateBillablePeopleResponse.LateBillablePerson
+                            {
+                                Id = slackProjectManager.Id,
+                                Email = slackProjectManager.Email
+                            });
+                        }
+                    }
+                }
+            }
+
+            return getProjectManagersWithOpenTimeEntriesResponse;
+        }
+
+        private IEnumerable<HarvestBillablePerson> HarvestProjectManagers(IEnumerable<HarvestBillablePerson> billablePeople, TimeSheet timeSheet)
+        {
+            return billablePeople.Where(person => timeSheet.ProjectManagerIds.Contains(person.Id));
+        }
+
+        private SlackBillablePerson SlackProjectManager(IList<SlackBillablePerson> slackBillablePeople, HarvestBillablePerson harvestProjectManager)
+        {
+            return slackBillablePeople.SingleOrDefault(billablePerson => String.Equals(RemoveTopLevelDomain(billablePerson.Email), RemoveTopLevelDomain(harvestProjectManager.Email), StringComparison.OrdinalIgnoreCase));
+        }
+
+        private static DateTimeOffset GetStartingDate(DateTimeOffset currentDateTime)
+        {
+            var daysFromMonday = (7 + (currentDateTime.DayOfWeek - DayOfWeek.Monday)) % 7;
+            
+            return currentDateTime.AddDays(-daysFromMonday);
+        }
+
+        private static DateTimeOffset GetEndingDate(DateTimeOffset currentDateTime)
+        {
+            if (IsEndOfTheMonth(currentDateTime)) return currentDateTime;
+            
+            var daysToFriday = (7 + (DayOfWeek.Friday - currentDateTime.DayOfWeek)) % 7;
+            
+            return currentDateTime.AddDays(daysToFriday);
+        }
```

Comment:
These methods hint at an abstraction for `BillingPeriod` or similar (the start and end date of the billing week, or partial week if at the start or end of a month).  They are duplicated from other classes too.

I haven't extracted to a helper class to avoid re-writing The Wolves because it would require changing several public interfaces to inject the new abstraction.

Reply:
Good thinking IMO. Nothing stops you from refactoring these methods later on.
A refactoring of this part of the code would make this PR *way* too big.
PS: these methods look like prime candidates for some characterisation tests 😁.